kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -562,6 +568,13 @@ def _init_and_run_proof(_init_problem: tuple[str, str]) -> tuple[bool, list[str]
             failure_log = None
             if not passed:
                 failure_log = print_failure_info(proof, kcfg_explore, counterexample_info)
+
+            vacuous_nodes = {node.id for node in proof.kcfg.nodes if _check_vacuous(node.cterm)}
```

Comment:
Constructing a `list` is cheaper, and `proof.kcfg.nodes` are already unique:

```suggestion
            vacuous_nodes = [node.id for node in proof.kcfg.nodes if _check_vacuous(node.cterm)]
```

Reply:
Done.